pkg/util/sqlkiller/sqlkiller_test.go

```
@@ -108,3 +110,82 @@ func TestSQLKillerConcurrentReset(t *testing.T) {
+// been observed the cancel hook must have run at least once, so an in-flight
+// RPC is always aborted instead of blocking until CoprReqTimeout.
+func TestConnCancelRaceSignalVsRegistration(t *testing.T) {
+	for range 200 {
+		killer := &SQLKiller{}
+		var calls atomic.Int32
+		cancel := func() { calls.Add(1) }
+
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		// Registration goroutine: install the hook via the production path, which
```

golangci-lint (2.12.2)

[warning] line-length-limit: line is 82 characters, out of limit 80

(revive)